compiler/passes/flattenFunctions.cpp

```diff
@@ -93,6 +93,11 @@ passableByVal(Type* type) {
 // Otherwise passing by value is more efficient.
 static bool
 passByRef(Symbol* sym) {
+
+  // Is it a non-POD record type?
+  if( sym->type->symbol->hasFlag(FLAG_NOT_POD) )
+    return true;
+
```

Comment:
I'm adding the comment 
+  // non-POD records are passed by reference here because
+  // other code probably won't copy them correctly if they
+  // are passed by value.